storage-bigtable/src/lib.rs

```diff
@@ -547,6 +547,67 @@ impl LedgerStorage {
         Ok(transaction_info.into())
     }
 
+    // Fetches and gets a vector of confirmed transactions via a multirow fetch
+    pub async fn get_confirmed_transactions(
+        &self,
+        signatures: &[Signature],
+    ) -> Result<Vec<ConfirmedTransactionWithStatusMeta>> {
+        debug!(
+            "LedgerStorage::get_confirmed_transactions request received: {:?}",
+            signatures
+        );
+        inc_new_counter_debug!("storage-bigtable-query", 1);
+        let mut bigtable = self.connection.client();
+
+        // Fetch transactions info
+        let keys = signatures.iter().map(|s| s.to_string()).collect::<Vec<_>>();
+        let cells = bigtable
+            .get_bincode_cells::<TransactionInfo>("tx", &keys)
+            .await?;
+
+        // Collect by slot
+        let mut slots: HashMap<Slot, HashMap<u32, String>> = Default::default();
+        for cell in cells {
+            if let (signature, Ok(TransactionInfo { slot, index, .. })) = cell {
+                slots.entry(slot).or_default().insert(index, signature);
+            }
+        }
+
+        // Fetch blocks and extract transactions
+        let keys = slots.keys().copied().collect::<Vec<_>>();
+        let data = self
+            .get_confirmed_blocks_with_data(&keys)
+            .await?
+            .zip(std::iter::repeat(&slots))
```

Comment:
I don't get why this is needed. Am I missing something?

Reply:
`filter_map` takes `FnMut` so we can not use `slots` multiple times :/
but actually, I can create a ref to `slots` which will be possible to use in `filter_map`!